dot/node.go

```diff
@@ -341,7 +341,7 @@ func NewNode(cfg *Config, ks *keystore.GlobalKeystore, stopFunc func()) (*Node,
 	}
 
 	if cfg.Global.NoTelemetry {
-		return node, nil
+		telemetry.GetInstance().KillInstance()
```

Comment:
it would be nicer to have `GetInstance()` return a `NoopInstance` instead of having `KillInstance()`, you can add a global package var to the `telemetry` package and use that in `GetInstance()`, eg `telemetry.Enabled = cfg.Global.NoTelemetry`

Reply:
Yes, this is nicer, I've updated.